lib/ticks-to-tree.js

```diff
@@ -99,71 +113,191 @@ function ticksToTree (ticks, options = {}) {
           type = 'WASM'
         }
       }
-
       if (type && type !== 'JS') name += ' [' + type + (kind ? ':' + kind : '') + ']'
-      return { S, name, value: 0, top: 0 }
+      return { S, name, type, value: 0, top: 0 }
     })
-
     labelInitFrames(stack, regExp)
 
-    addToMergedTree(stack.map(({ S, name, value, top }) => ({ S, name, value, top })))
+    addToMergedTree(trees, stack.map(({ S, name, type, value, top }) => ({ S, name, type, value, top })))
     // mutate original (save another loop over stack + extra objects)
-    addToUnmergedTree(stack)
+    addToUnmergedTree(trees, stack)
   })
 
-  function addToMergedTree (stack) {
-    var lastFrame = null
-    stack.forEach((frame, ix) => {
-      // TODO need an elegant way to place inlined frames into *other* stacks
-      // where the function has disappeared – for now we just remove inlined
-      // stacks from merged graphs
-      if (frame.S > 2) return // skip inlined
-      if (ix > 0) lastFrame.children = lastFrame.children || []
-      const children = (ix === 0) ? merged.children : lastFrame.children
-      const child = children.find(({ name }) => name === frame.name)
-
-      if (child === undefined) children.push(frame)
-      else frame = child
-
-      if (ix === stack.length - 1) frame.top++
-      if (ix === 0) merged.value += 1
-      frame.value++
-
-      lastFrame = frame
+  let loadedSourceMaps = {}
+  if (sourceMaps) {
+    const mapper = async (jsLibrary) => {
+      loadedSourceMaps = loadSourceMap(jsLibrary, sourceMaps, loadedSourceMaps)
+      return loadedSourceMaps
+    }
+    await pMap(Object.keys(jsLibraries), mapper, { concurrency: 16 })
+  }
+  if (sourceMaps || relativePath) {
+    Object.keys(trees).map((tree) => {
```

Comment:
this does not need to be a map, just a forEach. I would replace it with a for loop instead.